### PR TITLE
Feature/453827: Build 'Why are you cancelling this registration?' page

### PR DESCRIPTION
**Pull request by karthikeviden**
This includes changes for Grant, Cancel, Query and Reject

### CODE REVIEW

File: src/EPR.RegulatorService.Facade.API/Controllers/OrganisationRegistrationSubmissionsController.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+using EPR.RegulatorService.Facade.Core.Services.Submissions;
+using Microsoft.AspNetCore.Mvc;
+
+namespace EPR.RegulatorService.Facade.API.Controllers;
+
+[Route("api")]
+public class OrganisationRegistrationSubmissionsController(ISubmissionService submissionsService, ILogger<OrganisationRegistrationSubmissionsController> logger) : Controller
+{
+    private readonly ISubmissionService _submissionsService = submissionsService;
+    private readonly ILogger<OrganisationRegistrationSubmissionsController> _logger = logger;
```
Comment (prasadramireddyeviden) [inline src/EPR.RegulatorService.Facade.API/Controllers/OrganisationRegistrationSubmissionsController.cs:13]: Not required to define this class variables. Pl use prime constructor variables

File: src/EPR.RegulatorService.Facade.API/Controllers/OrganisationRegistrationSubmissionsController.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                RegulatorComment = request.Comments
+            },
+            User.UserId()
+        );
+
+        if (registrationSubmissionEvent.IsSuccessStatusCode) {
+            return Created();
+        }
+
+        _logger.LogWarning("Cannot create submission event");
```
[the remark this answers is not in the document]
Comment (karthikeviden) [reply src/EPR.RegulatorService.Facade.API/Controllers/OrganisationRegistrationSubmissionsController.cs:40]: After discussing with @JulianYoungEviden yesterday this was added as warning

File: src/EPR.RegulatorService.Facade.Core/Services/Submissions/SubmissionsService.cs [hunk 3]
```diff
@@ -4,19 +4,13 @@
...
 namespace EPR.RegulatorService.Facade.Core.Services.Submissions;
 
-public class SubmissionsService : ISubmissionService
+public class SubmissionsService(
+    HttpClient httpClient,
+    IOptions<SubmissionsApiConfig> options) : ISubmissionService
 {
-    private readonly HttpClient _httpClient;
-    private readonly SubmissionsApiConfig _config;
+    private readonly HttpClient _httpClient = httpClient;
```
[the remark this answers is not in the document]
Comment (karthikeviden) [reply src/EPR.RegulatorService.Facade.Core/Services/Submissions/SubmissionsService.cs:11]: This is already changed to primary constructor but can didn't remove the fields which I will do now, thanks

File: src/EPR.RegulatorService.Facade.API/Controllers/OrganisationRegistrationSubmissionsController.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (JulianYoungEviden) [reply src/EPR.RegulatorService.Facade.API/Controllers/OrganisationRegistrationSubmissionsController.cs:40]: I will take a look at this too....

**Review by JulianYoungEviden — DISMISSED** [review]

**Review by shehzad-ismail8 — DISMISSED** [review]

**Review by JulianYoungEviden — APPROVED** [review]

**Review by shehzad-ismail8 — APPROVED** [review]

**Pull request merged**